cray_product_catalog/catalog_delete.py

```diff
@@ -101,10 +109,14 @@ def modify_config_map(name, namespace, product, product_version, key=None):
             LOGGER.exception("Error calling read_namespaced_config_map")
 
             # ConfigMap doesn't exist yet
-            if err.status == 404 and attempt < max_attempts:
+            if err.status != ERR_NOT_FOUND and attempt > max_attempts:
+                raise   # unrecoverable
+            elif attempt == max_attempts and name != CONFIG_MAP:
+                LOGGER.info("ConfigMap %s/%s doesn't exist, so nothing to delete")
+                return
+            else:
                 LOGGER.warning("ConfigMap %s/%s doesn't exist, attempting again.", namespace, name)
                 continue
```

Comment:
Because the first if clause always raises an exception and the second clause always returns, this can be simplified.
Also, the `LOGGER.info` function call is missing the `namespace` and `name` arguments.
```suggestion
            if attempt == max_attempts and name != CONFIG_MAP:
                LOGGER.info("ConfigMap %s/%s doesn't exist, so nothing to delete", namespace, name)
                return
            LOGGER.warning("ConfigMap %s/%s doesn't exist, attempting again.", namespace, name)
            continue
```